pkg/bmt/reference/reference_test.go

```diff
@@ -98,12 +103,15 @@ func TestRefHasher(t *testing.T) {
 				return sha3hash(t, sha3hash(t, sha3hash(t, data[:64]), sha3hash(t, data[64:128])), sha3hash(t, sha3hash(t, data[128:192]), sha3hash(t, data[192:])))
 			},
 		},
-	} {
+	}
+
+	// run the tests
+	for _, x := range tests {
 		for segCount := x.from; segCount <= x.to; segCount++ {
 			for length := 1; length <= segCount*32; length++ {
 				t.Run(fmt.Sprintf("%d_segments_%d_bytes", segCount, length), func(t *testing.T) {
-					g := mockbytes.New(i, mockbytes.MockTypeStandard)
-					data, err := g.RandomBytes(length)
+					data := make([]byte, length)
+					_, err := io.ReadFull(crand.Reader, data)
```

Comment:
be aware that `mockbytes` generate random bytes with a reproducible seed. here you are switching to random tests which we cannot debug since we don't have the seed anymore